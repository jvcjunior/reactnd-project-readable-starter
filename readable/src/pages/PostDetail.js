import React, { Component } from 'react';
import * as actions from '../actions'

class PostDetail extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        const { dispatch, match } = this.props
        dispatch(actions.requestPostDetails(match.params.post_id))
    }

    render() {
        const { post } = this.props;
        return (
            <div> DETAILS </div>
        );
    }
}

export default PostDetail;