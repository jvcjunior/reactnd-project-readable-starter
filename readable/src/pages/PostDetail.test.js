import React from 'react';
import { shallow } from 'enzyme';
import PostDetail from './PostDetail';

describe('Post Detail Component', () => {
    const detail = shallow(<PostDetail />);

    it('renders post detail component correctly', ()=>{
        expect(detail).toMatchSnapshot();
    })

});