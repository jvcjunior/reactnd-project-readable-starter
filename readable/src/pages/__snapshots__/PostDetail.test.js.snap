// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Detail Component renders post detail component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostDetail />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " Details ",
    },
    "ref": null,
    "rendered": " Details ",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " Details ",
      },
      "ref": null,
      "rendered": " Details ",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Post Detail Component renders psot detail component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostDetail />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": " Details ",
    },
    "ref": null,
    "rendered": " Details ",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " Details ",
      },
      "ref": null,
      "rendered": " Details ",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
